src/components/Autocomplete/SectionItem/SectionItemText.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Item } from '../../../types';
+import { isInGroupSuggestion } from '../../../typeGuards';
+import { escapeRegExp } from '../../../utils';
+
+export interface SectionItemTextProps {
+  item: Item;
+  query: string;
+  highlightSearchTerm?: boolean;
+}
+
+export default function SectionItemText({
+  item,
+  query,
+  highlightSearchTerm,
+}: SectionItemTextProps) {
+  const itemText = isInGroupSuggestion(item) ? item.groupName : item.value;
+
+  if (highlightSearchTerm) {
+    const queryRegex = new RegExp(`(${escapeRegExp(query)})`, 'gi');
+    const queryIncludedInText = queryRegex.test(itemText);
+
+    if (query && queryIncludedInText) {
+      const splitText = itemText.split(queryRegex);
+
+      return (
+        <>
+          {splitText.map((split, index) => {
+            // eslint-disable-next-line react/no-array-index-key
+            if (queryRegex.test(split)) return <b key={`matched-word-${index}`}>{split}</b>;
```

Comment:
If you can make this an if else statement for readability it would be great. Also you can ignore this